src/api/controllers/AuthController.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  Body, Delete, JsonController, Params, Post,
+} from 'routing-controllers';
+import { UuidParam } from '../validators/GenericRequests';
+import { AuthService } from '../../services/AuthService';
+import {
+  CreateUserRequest,
+  ErrorResponse,
+  GetUserResponse,
+  getErrorMessage,
+} from '../../types';
+
+@JsonController('auth/')
+export class AuthController {
+  private authService: AuthService;
+
+  constructor(authService: AuthService) {
+    this.authService = authService;
+  }
+  
+  @Post()
+  async createUser(@Body() createUserRequest: CreateUserRequest): Promise<GetUserResponse | ErrorResponse> {
+    try {
+      const user = await this.authService.createUser(createUserRequest);
+      return { success: true, user: user };
+    } catch (error) {
+      return { success: false, error: getErrorMessage(error) }
+    }
+  }
+
+  @Delete('id/:id/')
```

Comment:
userId

Reply:
From what I'm aware, I can't call this `userId` unless I make a new validator in [`GenericRequests.ts`](https://github.com/cuappdev/resell-backend/blob/main/src/api/validators/GenericRequests.ts). But this would also mean I make one for `postId`, `userId`, etc. which feels unnecessary...